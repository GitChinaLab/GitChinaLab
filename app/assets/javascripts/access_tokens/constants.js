// Token types
export const FEED_TOKEN = 'feedToken';
export const INCOMING_EMAIL_TOKEN = 'incomingEmailToken';
export const STATIC_OBJECT_TOKEN = 'staticObjectToken';
