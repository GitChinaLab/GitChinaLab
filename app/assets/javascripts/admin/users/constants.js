import { s__, __ } from '~/locale';

export const USER_AVATAR_SIZE = 32;

export const LENGTH_OF_USER_NOTE_TOOLTIP = 100;

export const I18N_USER_ACTIONS = {
  edit: __('Edit'),
  userAdministration: s__('AdminUsers|User administration'),
  unlock: __('Unlock'),
  block: s__('AdminUsers|Block'),
  unblock: s__('AdminUsers|Unblock'),
  approve: s__('AdminUsers|Approve'),
  reject: s__('AdminUsers|Reject'),
  deactivate: s__('AdminUsers|Deactivate'),
  activate: s__('AdminUsers|Activate'),
  ldapBlocked: s__('AdminUsers|Cannot unblock LDAP blocked users'),
  delete: s__('AdminUsers|Delete user'),
  deleteWithContributions: s__('AdminUsers|Delete user and contributions'),
  ban: s__('AdminUsers|Ban user'),
  unban: s__('AdminUsers|Unban user'),
};

export const CONFIRM_DELETE_BUTTON_SELECTOR = '.js-delete-user-modal-button';

export const MODAL_TEXTS_CONTAINER_SELECTOR = '#js-modal-texts';

export const MODAL_MANAGER_SELECTOR = '#js-delete-user-modal';
