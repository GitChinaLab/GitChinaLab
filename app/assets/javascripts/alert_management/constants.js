import { s__ } from '~/locale';

export const SEVERITY_LEVELS = {
  CRITICAL: s__('severity|Critical'),
  HIGH: s__('severity|High'),
  MEDIUM: s__('severity|Medium'),
  LOW: s__('severity|Low'),
  INFO: s__('severity|Info'),
  UNKNOWN: s__('severity|Unknown'),
};

export const ALERTS_STATUS_TABS = [
  {
    title: s__('AlertManagement|Open'),
    status: 'OPEN',
    filters: ['TRIGGERED', 'ACKNOWLEDGED'],
  },
  {
    title: s__('AlertManagement|Triggered'),
    status: 'TRIGGERED',
    filters: 'TRIGGERED',
  },
  {
    title: s__('AlertManagement|Acknowledged'),
    status: 'ACKNOWLEDGED',
    filters: 'ACKNOWLEDGED',
  },
  {
    title: s__('AlertManagement|Resolved'),
    status: 'RESOLVED',
    filters: 'RESOLVED',
  },
  {
    title: s__('AlertManagement|All alerts'),
    status: 'ALL',
    filters: ['TRIGGERED', 'ACKNOWLEDGED', 'RESOLVED'],
  },
];

/* eslint-disable @gitlab/require-i18n-strings */

/**
 * Tracks snowplow event when user views alerts list
 */
export const trackAlertListViewsOptions = {
  category: 'Alert Management',
  action: 'view_alerts_list',
};
