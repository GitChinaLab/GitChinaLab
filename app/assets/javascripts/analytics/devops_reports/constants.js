import { __ } from '~/locale';

export const INTRO_COOKIE_KEY = 'dev_ops_report_intro_callout_dismissed';

export const INTRO_BANNER_TITLE = __('Introducing Your DevOps Reports');

export const INTRO_BANNER_BODY = __(
  'Your DevOps Reports give an overview of how you are using GitLab from a feature perspective. Use them to view how you compare with other organizations, and how your teams compare against each other.',
);

export const INTRO_BANNER_ACTION_TEXT = __('Read more');
