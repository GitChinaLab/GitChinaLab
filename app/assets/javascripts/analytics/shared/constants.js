import { masks } from 'dateformat';

export const DATE_RANGE_LIMIT = 180;
export const OFFSET_DATE_BY_ONE = 1;
export const PROJECTS_PER_PAGE = 50;

const { isoDate, mediumDate } = masks;
export const dateFormats = {
  isoDate,
  defaultDate: mediumDate,
  defaultDateTime: 'mmm d, yyyy h:MMtt',
  month: 'mmmm',
};
