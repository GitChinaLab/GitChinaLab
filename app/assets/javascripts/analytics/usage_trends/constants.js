import { getDateInPast } from '~/lib/utils/datetime_utility';

export const TOTAL_DAYS_TO_SHOW = 365;
export const TODAY = new Date();
export const START_DATE = getDateInPast(TODAY, TOTAL_DAYS_TO_SHOW);
