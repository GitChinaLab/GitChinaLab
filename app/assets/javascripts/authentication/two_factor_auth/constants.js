export const COPY_BUTTON_ACTION = 'copy';
export const DOWNLOAD_BUTTON_ACTION = 'download';
export const PRINT_BUTTON_ACTION = 'print';

export const TRACKING_LABEL_PREFIX = '2fa_recovery_codes_';

export const RECOVERY_CODE_DOWNLOAD_FILENAME = 'gitlab-recovery-codes.txt';

export const SUCCESS_QUERY_PARAM = 'two_factor_auth_enabled_successfully';

export const COPY_KEYBOARD_SHORTCUT = 'mod+c';
