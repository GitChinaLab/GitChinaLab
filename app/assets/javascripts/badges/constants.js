export const GROUP_BADGE = 'group';
export const PROJECT_BADGE = 'project';
