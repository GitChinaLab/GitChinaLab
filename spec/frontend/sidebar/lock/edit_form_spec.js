import { shallowMount } from '@vue/test-utils';
import EditForm from '~/sidebar/components/lock/edit_form.vue';
import { ISSUABLE_TYPE_ISSUE, ISSUABLE_TYPE_MR } from './constants';

describe('Edit Form Dropdown', () => {
  let wrapper;
  let issuableType; // Either ISSUABLE_TYPE_ISSUE or ISSUABLE_TYPE_MR
  let issuableDisplayName;

  const setIssuableType = (pageType) => {
    issuableType = pageType;
    issuableDisplayName = issuableType.replace(/_/g, ' ');
  };

  const findWarningText = () => wrapper.find('[data-testid="warning-text"]');

  const createComponent = ({ props }) => {
    wrapper = shallowMount(EditForm, {
      propsData: {
        isLocked: false,
        issuableDisplayName,
        ...props,
      },
    });
  };

  afterEach(() => {
    wrapper.destroy();
    wrapper = null;
  });

  describe.each`
    pageType
    ${ISSUABLE_TYPE_ISSUE} | ${ISSUABLE_TYPE_MR}
  `('In $pageType page', ({ pageType }) => {
    beforeEach(() => {
      setIssuableType(pageType);
    });

    describe.each`
      isLocked | lockStatusText
      ${false} | ${'unlocked'}
      ${true}  | ${'locked'}
    `('when $lockStatusText', ({ isLocked }) => {
      beforeEach(() => {
        createComponent({ props: { isLocked } });
      });

      it(`the appropriate warning text is rendered`, () => {
        expect(findWarningText().element).toMatchSnapshot();
      });
    });
  });
});
