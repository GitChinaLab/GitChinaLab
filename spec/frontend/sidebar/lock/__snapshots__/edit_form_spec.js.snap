// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Edit Form Dropdown In issue page when locked the appropriate warning text is rendered 1`] = `
<div
  class="dropdown-menu sidebar-item-warning-message"
  data-testid="warning-text"
>
  <p
    class="text"
  >
    <gl-sprintf-stub
      message="Unlock this %{issuableDisplayName}? %{strongStart}Everyone%{strongEnd} will be able to comment."
    />
  </p>
   
  <edit-form-buttons-stub
    islocked="true"
    issuabledisplayname="issue"
  />
</div>
`;

exports[`Edit Form Dropdown In issue page when unlocked the appropriate warning text is rendered 1`] = `
<div
  class="dropdown-menu sidebar-item-warning-message"
  data-testid="warning-text"
>
  <p
    class="text"
  >
    <gl-sprintf-stub
      message="Lock this %{issuableDisplayName}? Only %{strongStart}project members%{strongEnd} will be able to comment."
    />
  </p>
   
  <edit-form-buttons-stub
    issuabledisplayname="issue"
  />
</div>
`;

exports[`Edit Form Dropdown In merge request page when locked the appropriate warning text is rendered 1`] = `
<div
  class="dropdown-menu sidebar-item-warning-message"
  data-testid="warning-text"
>
  <p
    class="text"
  >
    <gl-sprintf-stub
      message="Unlock this %{issuableDisplayName}? %{strongStart}Everyone%{strongEnd} will be able to comment."
    />
  </p>
   
  <edit-form-buttons-stub
    islocked="true"
    issuabledisplayname="merge request"
  />
</div>
`;

exports[`Edit Form Dropdown In merge request page when unlocked the appropriate warning text is rendered 1`] = `
<div
  class="dropdown-menu sidebar-item-warning-message"
  data-testid="warning-text"
>
  <p
    class="text"
  >
    <gl-sprintf-stub
      message="Lock this %{issuableDisplayName}? Only %{strongStart}project members%{strongEnd} will be able to comment."
    />
  </p>
   
  <edit-form-buttons-stub
    issuabledisplayname="merge request"
  />
</div>
`;
